### PR TITLE
issue fixed 26106

### PR DESCRIPTION
**Pull request by sachinguptartpl**
### Description (*)

Downloadable Product should be OUT OF STOCK if there are no links just samples saved issue fixed
### Fixed Issues (if relevant)

1. magento/magento2#26106: Downloadable Product should be OUT OF STOCK if there are no links just samples saved

### Manual testing scenarios (*)

1. Add sample links only for downloadable product and check the result of graphql 
2. Add original link and sample link also for downloadable product and check the result of graphql 
3. Confirm the result of the GraphQl

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/CatalogInventoryGraphQl/Model/Resolver/StockStatusProvider.php [hunk 1]
```diff
@@ -48,6 +49,20 @@ public function resolve(Field $field, $context, ResolveInfo $info, array $value
...
+            if (!$downloadableTypeInstance->hasLinks($product)) {
+                $stockStatusByProductType = StockStatusInterface::STATUS_OUT_OF_STOCK;
+            } else {
+                $stockStatusByProductType = StockStatusInterface::STATUS_IN_STOCK;
+            }
+        }
+
+        return
+            $productStockStatus === StockStatusInterface::STATUS_IN_STOCK
+            && $stockStatusByProductType === StockStatusInterface::STATUS_IN_STOCK
```
Comment (lbajsarowicz) [inline app/code/Magento/CatalogInventoryGraphQl/Model/Resolver/StockStatusProvider.php:64]: Extract to private method `isDownloadableProductInStock` (add also line 50)

File: app/code/Magento/CatalogInventoryGraphQl/Model/Resolver/StockStatusProvider.php [hunk 2]
```diff
@@ -48,6 +49,20 @@ public function resolve(Field $field, $context, ResolveInfo $info, array $value
         $stockStatus = $this->stockStatusRepository->get($product->getId());
         $productStockStatus = (int)$stockStatus->getStockStatus();
 
-        return $productStockStatus === StockStatusInterface::STATUS_IN_STOCK ? 'IN_STOCK' : 'OUT_OF_STOCK';
```
Comment (lbajsarowicz) [inline app/code/Magento/CatalogInventoryGraphQl/Model/Resolver/StockStatusProvider.php:51]: You removed that line, but there's no value for it when product is not downloadable. So the `return` statement will be always false.

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Please find comments attached.

**Review by lenaorobei — CHANGES_REQUESTED** [review]
This issue should be fixed on the different layer, other than resolver. Please have a look at `getStockStatus`. Need to debug and understand, why it does not return correct information (does not take `saleable` into account).

**Comment by lenaorobei** [comment]
@sachinguptartpl, closing this PR due to inactivity. Please feel free to reopen if you are up to finishing it.

**Pull request closed without merging**